frontend/src/lib/components/GroupFilter.svelte

```diff
@@ -0,0 +1,80 @@
+<script>
+    import { items, selectedGroup } from '$lib/stores';
+    import { getGroupColor } from '$lib/util';
+
+    let groups = $derived(getUniqueGroups($items));
+
+    function getUniqueGroups(items) {
+        const groupMap = new Map();
+        
+        const allGroup = { uuid: null, name: "All" };
+        groupMap.set(null, allGroup);
+        
+        items.forEach(item => {
+            if (item.groups && item.groups.length > 0) {
+                item.groups.forEach(group => {
+                    if (!groupMap.has(group.uuid)) {
+                        const name = group.name || `Group ${group.uuid.substring(0, 8)}`;
+                        
+                        groupMap.set(group.uuid, { uuid: group.uuid, name });
+                    }
+                });
+            }
+        });
+        
+        // Convert the Map to an array and sort by name
+        return Array.from(groupMap.values()).sort((a, b) => {
+            if (a.uuid === null) return -1; // "All" always first
+            if (b.uuid === null) return 1;
+            return a.name.localeCompare(b.name);
+        });
+    }
```

Comment:
The  `getUniqueGroups()` can be removed and the `groups` variable can then be set with a call to `GetGroups()` during component mount.

Reply:
Oh indeed, I first thought we didn't have the groups object available in the frontend. Don't know why I thought this, fixed, thanks !